bin.src/convert_gen2_repo_to_gen3.py

```diff
@@ -0,0 +1,212 @@
+
+
+def convert(gen2root, gen3root, instrumentClass, calibFilterType,
+            skymapName=None, skymapConfig=None,
+            calibs=None, config=None):
+    """Convert the gen 2 Butler repo living at gen2root into a gen 3 repo
+    living at gen3root.
+
+    Parameters
+    ----------
+    gen2root : `str`
+        Root path to the gen2 repo to be converted.
+    gen3root : `str`
+        Root path to the gen3 output repo.
+    instrumentClass : `str`
+        Full python path to the `lsst.obs.base.Instrument` class of the repo
+        being converted.
+    calibFilterType : `str`
+        `abstract_filter` or `physical_filter`, depending on the type of
+        ``filter`` in the gen2 calib registry.
+    skymapName : `str`, optional
+        Name of the skymap to be converted in the repo.
+    skymapConfig : `str`, optional
+        Config of the gen2 skymap to be converted.
+    calibs : `str`, optional
+        Path to the gen2 calibration repository to be converted.
+        If a relative path, it is assumed to be relative to ``gen2root``.
+    config : `str`, optional
+        Additional configuration to load after all default/instrument
+        configurations.
+    """
+
+    # instantiate the correct instrument
+    obspkg, Instrument = instrumentClass.rsplit('.', maxsplit=1)
+    instrument = getattr(importlib.import_module(obspkg), Instrument)()
+
+    lsst.daf.butler.Butler.makeRepo(gen3root)
+
+    convertRepoConfig = ConvertRepoTask.ConfigClass()
+    instrument.applyConfigOverrides(ConvertRepoTask._DefaultName, convertRepoConfig)
+    if convertRepoConfig.raws.instrument is None:
+        convertRepoConfig.raws.instrument = instrumentClass
+    convertRepoConfig.raws.transfer = "symlink"
+    if skymapName is not None:
+        convertRepoConfig.skyMaps[skymapName] = ConvertRepoSkyMapConfig()
+        convertRepoConfig.skyMaps[skymapName].load(skymapConfig)
+    if config is not None:
+        convertRepoConfig.load(config)
+
+    configure_translators(instrument, calibFilterType, convertRepoConfig.ccdKey)
+
+    butlerConfig = lsst.daf.butler.ButlerConfig(other=gen3root)
+    butler = lsst.daf.butler.Butler(butlerConfig, run=instrument.getName())
+    convertRepoTask = ConvertRepoTask(config=convertRepoConfig, butler3=butler)
+    convertRepoTask.run(
+        root=gen2root,
+        # NOTE: we'd like to use `raw/NAME` and `calib/NAME` here, but if I do, I get an error about
+        # `AssertionError: Multiple collections for curated calibrations is not yet supported.`
+        # since we also want to specify just `NAME` so that we can instantiate a gen3 Butler
+        # using that as an "umbrella" collection so it knows about everything in it.
+        collections=[instrument.getName()],  # ['raw/'+instrument.getName(), instrument.getName()],
+        calibs=None if calibs is None else {calibs: [instrument.getName()]}
+        # ['calib/HSC'+instrument.getName(), instrument.getName()]},
```

Comment:
I think these can safely go, as the commented-out code would be easy to reconstruct later.  I'd be more sympathetic to the commented-out translator rules sticking around, as I think some of those may need to come back (possibly in some other package) quite soon, but I think I'd still just stick with the general rule that git history is the right place to look for dead code, rather than comments.

Reply:
The one unique translator lives on in HSC.